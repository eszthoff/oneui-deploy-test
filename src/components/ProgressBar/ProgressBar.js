import React from 'react';
import PropTypes from 'prop-types';
import bem from '../../packages/bem';
import styles from './ProgressBar.scss';
import { CONTEXTS } from '../../constants';

const { block, elem } = bem({
    name: 'ProgressBar',
    classnames: styles,
    propsToMods: ['animated', 'context', 'hidden']
});

const ProgressBar = props => {
    const { context, label, percentage, ...rest } = props;
    return (
        <div {...rest} {...block(props)}>
            <div
                {...elem('fill', props)}
                style={{
                    width: `${percentage}%`
                }}
            >
                {label || `${Number(percentage)}%`}
            </div>
        </div>
    );
};

ProgressBar.propTypes = {
    /** Show progress activity with animation */
    animated: PropTypes.bool,
    /** The progress bar context (e.g. brand, primary, bad, good etc. - defaults to brand) */
    context: PropTypes.oneOf(CONTEXTS),
    /** Hides the progress bar if true */
    hidden: PropTypes.bool,
    /** Text to show instead of percentage */
    label: PropTypes.node,
    /** Percentage of progress bar to be filled */
    percentage: PropTypes.number.isRequired
};

ProgressBar.defaultProps = {
    animated: false,
    context: 'brand',
    hidden: false,
    label: null
};

export default ProgressBar;
