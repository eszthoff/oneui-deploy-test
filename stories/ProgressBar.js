import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { boolean, number, select, text, withKnobs } from '@storybook/addon-knobs'; // eslint-disable-line import/no-extraneous-dependencies
import ProgressBar from '../src/components/ProgressBar';
import { CONTEXTS } from '../src/constants';

storiesOf('ProgressBar', module)
    .addDecorator(withKnobs)
    .add('ProgressBar', () => (
        <ProgressBar
            animated={boolean('Animated', true)}
            context={select('Context', CONTEXTS, CONTEXTS[1])}
            hidden={boolean('Hidden', false)}
            label={text('Label', 'Loading...')}
            percentage={number('Percentage', 50)}
        />
    ));
