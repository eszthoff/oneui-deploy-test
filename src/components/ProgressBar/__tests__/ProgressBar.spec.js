import React from 'react';
import toJson from 'enzyme-to-json';
import ProgressBar from '../ProgressBar';

describe('<ProgressBar> that renders a horizontal progress bar', () => {
    it('should render a default progress bar correctly', () => {
        const wrapper = shallow(<ProgressBar percentage={25} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a progress bar with label', () => {
        const wrapper = shallow(<ProgressBar percentage={25}>Loading...</ProgressBar>);

        // Check that label is shown
        expect(wrapper.childAt(0).text()).toBe('Loading...');
    });

    it('should add classes when props are changed', () => {
        const wrapper = shallow(
            <ProgressBar percentage={25} context="primary" animated hidden small />
        );

        // Check that bar is animated
        expect(wrapper.hasClass('ProgressBar--animated')).toBe(true);
        expect(wrapper.childAt(0).hasClass('ProgressBar__fill--animated')).toBe(true);

        // Check that bar is hidden
        expect(wrapper.hasClass('ProgressBar--hidden')).toBe(true);
        expect(wrapper.childAt(0).hasClass('ProgressBar__fill--hidden')).toBe(true);

        // Check that bar is small
        expect(wrapper.hasClass('ProgressBar--small')).toBe(true);
        expect(wrapper.childAt(0).hasClass('ProgressBar__fill--small')).toBe(true);

        // Check that context is primary
        expect(wrapper.hasClass('ProgressBar--context_primary')).toBe(true);
        expect(wrapper.childAt(0).hasClass('ProgressBar__fill--context_primary')).toBe(true);
    });
});
