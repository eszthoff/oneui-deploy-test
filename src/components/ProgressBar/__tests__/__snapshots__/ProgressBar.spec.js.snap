// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar> that renders a horizontal progress bar should add classes when props are changed 1`] = `
<div
  animated={true}
  className="ProgressBar ProgressBar--animated ProgressBar--context_primary ProgressBar--hidden ProgressBar--small"
  hidden={true}
>
  <div
    className="ProgressBar__fill ProgressBar__fill--animated ProgressBar__fill--context_primary ProgressBar__fill--hidden ProgressBar__fill--small"
    style={
      Object {
        "width": "25%",
      }
    }
  />
</div>
`;

exports[`<ProgressBar> that renders a horizontal progress bar should render a default progress bar correctly 1`] = `
<div
  animated={false}
  className="ProgressBar ProgressBar--context_brand"
  hidden={false}
>
  <div
    className="ProgressBar__fill ProgressBar__fill--context_brand"
    style={
      Object {
        "width": "25%",
      }
    }
  >
    25%
  </div>
</div>
`;

exports[`<ProgressBar> that renders a horizontal progress bar should render a progress bar with label 1`] = `
<div
  animated={false}
  className="ProgressBar ProgressBar--context_brand"
  hidden={false}
>
  <div
    className="ProgressBar__fill ProgressBar__fill--context_brand"
    style={
      Object {
        "width": "25%",
      }
    }
  >
    Loading...
  </div>
</div>
`;
